### PR TITLE
Add TemporaryContentAuthorGroup Rule

### PR DESCRIPTION
**Pull request by absarasw**

### CODE REVIEW

**Comment by absarasw** [comment]
@dulvac , @volteanu As discussed in https://github.com/adobe/aem-test-samples/pull/36 , we require to create a content author group in order to run the test. This PR will added the rule to create temporary content author group. Please review the PR for the same.

File: src/main/java/com/adobe/cq/testing/junit/rules/TemporaryContentAuthorGroup.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    }
+
+    /**
+     * Create unique authorizable Id to make sure no side effects from versioning / restore occurs
+     *
+     * @param authorizableId authorizable id
+     * @return unique authorizableId
+     */
+    protected String createUniqueAuthorizableId(String authorizableId) {
+        return authorizableId + new Date().getTime();
```
Comment (dulvac) [inline src/main/java/com/adobe/cq/testing/junit/rules/TemporaryContentAuthorGroup.java]: please use a UUID here instead or in addition to the timestamp

File: src/main/java/com/adobe/cq/testing/junit/rules/TemporaryContentAuthorGroup.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                // set permissions
+                permissionsObj.changePermissions(group.getId(), CONTENT_NODE, true, true, true, false,
+                        false, false, false, 200);
+                return true;
+            }
+        }
+
+        CreateGroupPolling p = new CreateGroupPolling();
+        try {
+            p.poll(SECONDS.toMillis(20), SECONDS.toMillis(1));
```
Comment (dulvac) [inline src/main/java/com/adobe/cq/testing/junit/rules/TemporaryContentAuthorGroup.java]: It would be nice to have the seconds as a configurable parameter.

**Comment by absarasw** [comment]
@dulvac I have updated the PR with your suggestions. Please check.

File: src/main/java/com/adobe/cq/testing/junit/rules/TemporaryContentAuthorGroup.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                // set permissions
+                permissionsObj.changePermissions(group.getId(), CONTENT_NODE, true, true, true, false,
+                        false, false, false, 200);
+                return true;
+            }
+        }
+
+        CreateGroupPolling p = new CreateGroupPolling();
+        try {
+            p.poll(DEFAULT_TIMEOUT, SECONDS.toMillis(DEFAULT_RETRY_DELAY));
```
Comment (dulvac) [inline src/main/java/com/adobe/cq/testing/junit/rules/TemporaryContentAuthorGroup.java:112]: Sorry, I meant making this configurable as a field on the `CreateGroupPolling` class and propagate that from a field on `TemporaryContentAuthorGroup`. Basically being able to control the total wait time from the caller.
e.g. 
` new TemporaryContentAuthorGroup(supplier).withTimeout(Duration.ofSeconds(30))`

**Comment by volteanu** [comment]
@absarasw I've merged this as it is of now to make a release. Feel free to follow-up with another PR for the improvements suggested by @dulvac

**Comment by volteanu** [comment]
released in 0.8.0

**Comment by absarasw** [comment]
@volteanu, thanks a lot. 
I'll open a follow up ticket for the improvements.

**Pull request merged**